### PR TITLE
router: remove send_local_reply_when_no_buffer_and_upstream_request guard

### LINKED ISSUE #36608: envoy_reloadable_features_quic_upstream_reads_fixed_number_packets deprecation
Your change #32693 (quic: upstream connections read 32 packets per loop) introduced a runtime guarded feature. It has been 6 months since the new code has been exercised by default, so it's time to remove the old code path. This issue tracks source code cleanup so we don't forget.

### LINKED ISSUE #36609: envoy_reloadable_features_send_local_reply_when_no_buffer_and_upstream_request deprecation
Your change #32336 (route: fix a timing issue and remove assertion.) introduced a runtime guarded feature. It has been 6 months since the new code has been exercised by default, so it's time to remove the old code path. This issue tracks source code cleanup so we don't forget.

### PR DESCRIPTION
**Pull request by botengyao**
Commit Message: router: remove send_local_reply_when_no_buffer_and_upstream_request guard
Risk Level: low
Testing: unit test
Docs Changes:
Fixes #36609

### CODE REVIEW

File: changelogs/current.yaml [hunk 1]
```diff
@@ -15,3 +15,6 @@ removed_config_or_runtime:
 new_features:
 
 deprecated:
+- area: router
```
Comment (alyssawilk) [inline changelogs/current.yaml:18]: I think this goes under removed config or runtime

**Review by alyssawilk — COMMENTED** [review]
code LG, 
/wait on release note

File: changelogs/current.yaml (hunk 1 above)
Comment (botengyao) [reply changelogs/current.yaml]: good catch, done.

**Review by alyssawilk — DISMISSED** [review]

**Review by alyssawilk — DISMISSED** [review]

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**